package.json

```diff
@@ -19,7 +19,7 @@
   ],
   "scripts": {
     "build": "tsc && npm run prettify && npm run lint",
-    "lint": "node ./bin/balena-lint lib/",
+    "lint": "node ./bin/balena-lint -t tsconfig.json lib/",
     "prepublish": "require-npm4-to-publish",
     "prepublishOnly": "npm run test",
     "prettify": "node ./bin/balena-lint --fix lib/",
```

Comment:
`balena-lint` is called in two scripts: `lint` and `prettify`. I have only added the `-t` option to `lint`, because the option has a performance penalty and it is redundant to do it twice. This means, however, that `Warning: The 'no-floating-promises' rule requires type information` is still printed when `prettify` is executed.

Reply:
How much of a performance penalty is it?

Reply:
On my Mac laptop:

Command | Time (lowest of multiple executions)
--- | ---
`node ./bin/balena-lint lib/` | 1.4s
`node ./bin/balena-lint -t tsconfig.json lib/` | 2.7s

Your question tells me (?) that you'd rather face the extra 1.3s than see the warning message, :-) so I am now adding the `'-t tsconfig.json'` option to both script lines.

In the case of `balena-cli`, being a much larger project:

Command | Time (lowest of multiple executions)
--- | ---
without `-t tsconfig.json` | 8.9s
with `-t tsconfig.json` | 14.8s

IMPORTANTLY, the performance penalty is only incurred if the `'-t'` option is used. Users who would rather `balena-lint` ran faster than taking advantage of 'no-floating-promises' can simply not use the `'-t'` flag, and the warning can be suppressed by overriding the default rules as explained in the README file. For `balena-cli`, I would rather take the 6s penalty than risking bugs of improperly handled promises.

Reply:
Actually it's more that I'd like to see it definitely get handled and was wondering on whether we could just enable it always. Do you know what typescript version that is using since I know typescript has been doing lots of optimizations and maybe with ts 4.5 it's feasible to enable it by default

Reply:
`tslint` is using the version of TypeScript installed as a peer dependency, subject to:
```
  "peerDependencies": {
    "typescript": ">=2.3.0-dev || >=2.4.0-dev || >=2.5.0-dev || >=2.6.0-dev || >=2.7.0-dev || >=2.8.0-dev || >=2.9.0-dev || >=3.0.0-dev || >= 3.1.0-dev || >= 3.2.0-dev || >= 4.0.0-dev"
  },
```
'no-floating-promises' is already enabled by default, but the user needs to specify the path to their relevant `tsconfig.json` file, which isn't necessarily named `tsconfig.json` and isn't necessarily located at the project root. For example, in balena-cli PR balena-io/balena-cli/pull/2389, I am specifying an alternative `-t tsconfig.dev.json` file. 

Most importantly, enabling type checking with `-t tsconfig.json` may produce errors that take some debugging to sort. If `balena-lint` enabled it by default by assuming `tsconfig.json` at the project root, then when users bumped `@balena/lint` in their projects, they might get error messages that they might not readily understand. The balena-cli experience is an example. Before I made the `package.json` adjustments in PR balena-io/balena-cli/pull/2389, I was getting errors when executing `balena-lint`. The 'no-floating-promises' rule was unhappy about checking files that did not have the `'.ts'` extension, such as:

```
balena-lint -e ts -e js -t tsconfig.dev.json --fix bin/balena bin/balena-dev completion/ gulpfile.js .mocharc.js .mocharc-standalone.js
```

So I had to refactor the lint npm script lines, for example:

```json
    "lint": "npm run lint-tsconfig && npm run lint-other",
    "lint-tsconfig": "balena-lint -e ts -e js -t tsconfig.dev.json --fix automation/ lib/ tests/ typings/",
    "lint-other": "balena-lint -e ts -e js --fix bin/balena bin/balena-dev completion/ gulpfile.js .mocharc.js .mocharc-standalone.js",
```

Using the `-t` flag only in some lint script lines, not in all of them.

So, fully enabling this by default would not be plain sailing for everyone. I suppose we could call it a breaking change and do a major version bump and add even more documentation to README, documenting the balena-cli experience as an example of how to handle the potential impact...

How about this: For now, we just merge this PR with `Change-type: minor`. Some people like you and the reviewers of this PR and maybe the supervisor team could try and see how it goes for their projects. If everybody is pleased, another PR, perhaps with `Change-type: major`, could enable it by default by assuming `tsconfig.json` at the project root and perhaps introducing `-t no` as a way of disabling it.

Reply:
I will merge this PR with the reasoning above, and further work and discussion towards making `-t tsconfig.json` a default behavior (always enabled) can take place in other PRs. Thank you for the review @Page- 🙏